diffsims/generators/diffraction_generator.py

```diff
@@ -244,26 +253,36 @@ def calculate_ed_data(
         z_sphere = -np.sqrt(r_sphere ** 2 - r_spot ** 2) + r_sphere
         excitation_error = z_sphere - z_spot
 
+        # determine the pre-selection reflections
         if self.precession_angle == 0:
-            # Mask parameters corresponding to excited reflections.
             intersection = np.abs(excitation_error) < max_excitation_error
-            intersection_coordinates = cartesian_coordinates[intersection]
-            excitation_error = excitation_error[intersection]
-            r_spot = r_spot[intersection]
-            g_indices = g_indices[intersection]
-            g_hkls = g_hkls[intersection]
-
+        else:
+            # only consider points that are between or touch the cones created by the ewald sphere extremes
+            # ewald sphere is approximated flat
+            intersection = np.abs(z_spot) - np.abs(
```

Comment:
I'm nervous about introducing a flat ewald sphere when we don't have to; is the speed loss is crippling?

Reply:
For very simple structures no, but once you have thousands of reciprocal space points for more complex structures (which I had) it makes a big difference. If the ewald sphere is not flat and one is afraid of cutting out too many reflections, max_excitation_error can just be increased and one can use shape_factor_width independently to control spot intensity. Spots that are incorrectly still included may be removed then using the min_intensity. So in my view there is no issue of either missing spots or additional spots. But I'll check whether a curved Ewald sphere can be taken into consideration nonetheless, shouldn't be too much more complicated.

Reply:
I've now updated the filter to take into consideration the ewald sphere curvature